src/main/java/net/dv8tion/jda/bot/entities/ApplicationInfo.java

```diff
@@ -0,0 +1,144 @@
+/*
+ *     Copyright 2015-2017 Austin Keener & Michael Ritter
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.bot.entities;
+
+import java.util.Collection;
+import java.util.List;
+import net.dv8tion.jda.core.JDA;
+import net.dv8tion.jda.core.Permission;
+import net.dv8tion.jda.core.entities.ISnowflake;
+import net.dv8tion.jda.core.entities.User;
+
+/**
+ * Represents a Discord Application from it's bot's point of view.
+ * 
+ * @since  JDA 3.0
+ * @author Aljoscha Grebe
+ * 
+ * @see {@link net.dv8tion.jda.bot.JDABot#getApplicationInfo() JDABot#getApplicationInfo()}
+ */
+public interface ApplicationInfo extends ISnowflake
+{
+
+    /**
+     * Whether the bot requires code grant to invite or not. 
+     * If your application requires multiple scopes then you may need the full OAuth2 flow to ensure a
+     * bot doesn't join before your application is granted a token.
+     * 
+     * @return whether the bot requires code grant
+     */
+    boolean doesBotRequireCodeGrant();
```

Comment:
Code grant in this case means that additional oauth2 steps are required to authorize the application to make a bot join a guild
Like `&response_type=code` together with a valid `&redirect_uri`